### PR TITLE
Add support for Ninja generator

### LINKED ISSUE #1126: Build fails with Ninja
## Description

For CMake-based projects, ninja is increasingly the preferred replacement for make at least on Linux, for various reasons, and attempting to build RenderDoc with it currently simply fails.

## Repro steps

1. Configure cmake as usual, adding the `-GNinja` option to select the Ninja generator (as opposed to the Makefile generator)
2. Run `ninja` instead of `make`.
3. Observe the error: 
>   ninja: error: build.ninja:2690: bad $-escape (literal $ must be written as $$)

## Environment

* Ubuntu 18.04 with the standard development packages (cmake version 3.10.2, ninja version 1.8.2)

### PR DESCRIPTION
**Pull request by darksylinc**
Fixes baldurk/renderdoc#1126

## Description

There's not much to say. It adds support for Ninja generator without breaking the rest.

### CODE REVIEW

File: qrenderdoc/CMakeLists.txt [hunk 1]
```diff
@@ -84,15 +84,21 @@ if(APPLE)
     set(SWIG_CONFIGURE_CXX "")
 endif()
 
+set( GENERATOR_MAKE "$(MAKE)" )
+if( CMAKE_MAKE_PROGRAM MATCHES "ninja$" )
+    set( GENERATOR_MAKE "make" )
+    message( ${GENERATOR_MAKE} )
```
Comment (baldurk) [inline qrenderdoc/CMakeLists.txt]: If this message was left over from debugging it should be removed, or if you wanted to leave it in then the message should be more descriptive like "Overriding ninja generator to make for subprojects" e.g.

**Comment by baldurk** [comment]
This seems fine - I've left a minor note.

File: qrenderdoc/CMakeLists.txt (hunk 1 above)
Comment (darksylinc) [reply qrenderdoc/CMakeLists.txt]: Oh my!!!
I was sure I cleaned that up!!!

I'll make a new PR with that gone and have the qt creator stuff split

**Comment by darksylinc** [comment]
I've fixed this PR. The other one is still pending a fix yet.

**Comment by baldurk** [comment]
Merged this in - though it does seem like this will still be non-ideal for ninja users because the `make` for qrenderdoc won't be parallel?

**Comment by darksylinc** [comment]
That is correct.
I didn't want to lose my time seeing if the custom step could run or be ported to ninja. I just wanted to fix that cryptic error message that makes building impossible on IDEs that default to using Ninja (making it troublesome to switch to make)

Indeed there is possibly more work that could be done. Or we could also try to deduce the amount of cores and add -jN (where N is number of cores) to make it parallel.

**Pull request merged**